pkg/collector/corechecks/cluster/kubernetes_apiserver_test.go

```diff
@@ -88,11 +88,8 @@ func TestParseComponentStatus(t *testing.T) {
 	}
 
 	// FIXME: use the factory instead
-	kubeASCheck := &KubeASCheck{
-		instance:              &KubeASConfig{},
-		CheckBase:             core.NewCheckBase(kubernetesAPIServerCheckName),
-		KubeAPIServerHostname: "hostname",
-	}
+	kubeASCheck := NewKubeASCheck(core.NewCheckBase(kubernetesAPIServerCheckName), &KubeASConfig{})
+	kubeASCheck.KubeAPIServerHostname = "hostname"
```

Comment:
do you think it makes sens to have also the `hostname` part of the `NewKubeASCheck()` args?

Reply:
Hm.. perhaps. Although I don't see the `KubeAPIServerHostname` being set in the check (although I see it used in the component status submission). I must be missing something..?

Reply:
Yes, I'm leaning towards removing it - I think this is the original PR that added it https://github.com/DataDog/datadog-agent/pull/983/files where it wasn't set outside of the tests